### ISSUE TITLE
ValueError: set_wakeup_fd only works in main thread

### ISSUE THREAD

**Issue by d3ned1** [GrahamDumpleton/mod_wsgi#616]
Hi, Mr. Dumpleton!

I opened this [issue](https://github.com/django/asgiref/issues/143) for **asgiref** long time ago, but we could not find the way to solution. Could you please take a look at our problem?
We think it is related to mod_wsgi.
You can find there tracebacks and our "research" as well.

Thank you!

**Comment by GrahamDumpleton** [comment]
Commented on other issue. You could have just tagged me in the other issue instead of creating issue here.

**Comment by d3ned1** [comment]
Thank you for your comments!

**Comment by luzik** [comment]
This bug is more and more painful. Can't wait for fix release!

**Comment by GrahamDumpleton** [comment]
To summarise here what the issue is, technically it is caused by a failing in the Python threading module in that Python doesn't actually initialize properly its record of what the main thread is, instead it assumes that the first thread to import the threading module is the main thread, which it may not be in embedded systems. So a proper fix would be required in the Python core runtime, although core developers may argue it isn't their problem so hasn't been reported.

A workaround is to force preloading of the WSGI script file in mod_wsgi. The preloading occurs in the main thread, so results then in the record of what is the main thread being correct.

To force preloading, you can use on Linux:

```
WSGIScriptAlias / /some/path/wsgi.py process-group=xxx application-group=%{GLOBAL}
```

and on Windows:

```
WSGIScriptAlias / /some/path/wsgi.py application-group=%{GLOBAL}
```

Appropriate values for ``process-group`` and ``application-group`` should be used based on your configuration.

When you use the ``process-group`` and ``application-group`` options with ``WSGIScriptAlias`` you do not need to set ``WSGIProcessGroup`` and ``WSGIApplicationGroup`` directives as the preloading using those options achieves the same result.

A workaround will also be included as part of mod_wsgi 4.8.0 when released.

**Comment by Wissperwind** [comment]
So if I need to bind my wsgi application to another endpoint than `/`, I can't use WSGIScriptAlias.
So I use Alias instead.  How can I add the application-group=%{GLOBAL} in that case?
